src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/SendOrderContestedAboutToSubmitHandler.java

```diff
@@ -55,139 +56,302 @@ public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType
+            latestAdditionalHearingDocument.ifPresent(
+                orders::add);
+        }
+
+        List<CaseDocument> otherHearingDocuments = documentHelper.getHearingDocumentsAsPdfDocuments(caseDetails, authorisationToken);
+        if (!otherHearingDocuments.isEmpty()) {
+            orders.addAll(otherHearingDocuments);
         }
+        return orders;
     }
 
-    private List<BulkPrintDocument> createHearingDocumentPack(Map<String, Object> caseData, String authorisationToken, String caseId) {
-        List<BulkPrintDocument> hearingDocumentPack = new ArrayList<>();
 
-        documentHelper.getDocumentLinkAsBulkPrintDocument(caseData, CONTESTED_ORDER_APPROVED_COVER_LETTER).ifPresent(hearingDocumentPack::add);
-        documentHelper.getDocumentLinkAsBulkPrintDocument(caseData, LATEST_DRAFT_HEARING_ORDER).ifPresent(hearingDocumentPack::add);
+    private void shareAndSendGeneralOrderWithSelectedParties(FinremCaseDetails caseDetails,
+                                                             List<String> partyList,
+                                                             DynamicMultiSelectList selectedOrders,
+                                                             List<OrderSentToPartiesCollection> printOrderCollection) {
 
-        if (documentHelper.hasAnotherHearing(caseData)) {
-            Optional<CaseDocument> latestAdditionalHearingDocument = documentHelper.getLatestAdditionalHearingDocument(caseData);
-            latestAdditionalHearingDocument.ifPresent(
-                caseDocument -> hearingDocumentPack.add(documentHelper.getCaseDocumentAsBulkPrintDocument(caseDocument)));
+        Long caseId = caseDetails.getId();
+        log.info("Share selected 'GeneralOrder' With selected parties for caseId {}", caseId);
+
+        FinremCaseData caseData = caseDetails.getData();
+        CaseDocument generalOrder = caseData.getGeneralOrderWrapper().getGeneralOrderLatestDocument();
+
+        if (generalOrderService.isSelectedOrderMatches(selectedOrders, generalOrder)) {
+            shareAndPrintGeneralOrderWithApplicant(caseDetails, partyList, generalOrder);
+            shareAndPrintGeneralOrderWithRespondent(caseDetails, partyList, generalOrder);
+            shareAndPrintOrderWithIntervenerOne(caseDetails, partyList, generalOrder);
+            shareAndPrintOrderWithIntervenerTwo(caseDetails, partyList, generalOrder);
+            shareAndPrintOrderWithIntervenerThree(caseDetails, partyList, generalOrder);
+            shareAndPrintOrderWithIntervenerFour(caseDetails, partyList, generalOrder);
+            printOrderCollection.add(addToPrintOrderCollection(generalOrder));
         }
+    }
 
-        List<BulkPrintDocument> otherHearingDocuments = documentHelper.getHearingDocumentsAsBulkPrintDocuments(
-            caseData, authorisationToken, caseId);
+    private void shareAndPrintOrderWithIntervenerFour(FinremCaseDetails caseDetails,
```

Comment:
Refactor these methods into a class to avoid repetitive code here if we can

Reply:
You have done this. Thank you.